File: Check if Array is Good.kt
class Solution {
    fun isGood(nums: IntArray): Boolean {
        val n = nums.size
        val bb = BooleanArray(n + 1)
        for (m in nums)
            if (m >= n || bb[m])
                if (m != n - 1 || bb[0]) return false
                else bb[0] = true
            else bb[m] = true
        return true
    }
}

File: check-if-array-is-good.py
class Solution:
    def isGood(self, nums: List[int]) -> bool:
        if len(nums) < 2:
            return False

        nums.sort()

        n = len(nums) - 1

        for i in range(n - 1):
            if nums[i] != i + 1:
                return False

        return nums[n - 1] == n and nums[n] == n